pkg/container/bubblewrap_runner.go

```diff
@@ -85,9 +85,16 @@ func (bw *bubblewrap) cmd(ctx context.Context, cfg *Config, debug bool, envOverr
 		"--chdir", runnerWorkdir,
 		"--clearenv")
 
+	// Set caps as for the current thread bounding capability set.
+	baseargs = append(baseargs, "--cap-add", "all")
+
+	baseargs = append(baseargs, "--unshare-user")
 	if cfg.RunAs != "" {
-		baseargs = append(baseargs, "--unshare-user")
 		baseargs = append(baseargs, "--uid", cfg.RunAs)
+		baseargs = append(baseargs, "--gid", cfg.RunAs)
+	} else {
+		baseargs = append(baseargs, "--uid", "0")
+		baseargs = append(baseargs, "--gid", "0")
 	}
```

Comment:
I am pondering about this. Many invocations of bubblewrap typically bind-mount read-only etc/passwd and etc/group such that things like this just work correctly.

Or better in the user-namespace create dedicated builder account to run.

Would that also fix many things?

Reply:
From what I see it just creates the account for build with uid/gid 1000 in passwd/group, with the apko config - that's why I was expecting to always be 1000 in bwrap when not specifyung any runAs.

Do you have some reference to those bind mounts so I can give a check?

These options (--uid/--gid=0) just map the current user id of the user namespace of the process that runs melange to 0 inside de bwrap user namespace (rootless) - uid_maps/gid_maps in procfs.

---

Whether we want to continue with rootless and applying least privilege approach I think it's fundamental, even though we should avoid sudo that it's causing inconsistent results IMO. That's the main point of this PR